core/store/src/trie/shard_tries.rs

```diff
@@ -333,6 +367,59 @@ impl ShardTries {
         let (_store, manager) = self.get_state_snapshot(&sync_prev_prev_hash)?;
         Ok(manager.get_flat_storage_status(shard_uid))
     }
+
+    /// Should be called upon startup to load in-memory tries for enabled shards.
+    pub fn load_mem_tries_for_enabled_shards(
+        &self,
+        shard_uids: &[ShardUId],
+    ) -> Result<(), StorageError> {
+        let trie_config = &self.0.trie_config;
+        let shard_uids_to_load = shard_uids
+            .iter()
+            .copied()
+            .filter(|shard_uid| {
+                trie_config.load_mem_tries_for_all_shards
+                    || trie_config.load_mem_tries_for_shards.contains(shard_uid)
+            })
+            .collect::<Vec<_>>();
+        let store = self.0.store.clone();
+        info!(target: "memtrie", "Loading tries to memory for shards {:?}...", shard_uids);
```

Comment:
Same as in previous PR - please look over all debug lines and use structured logging as much as possible 🙏